core/src/main/scala/neotypes/Transaction.scala

```diff
@@ -130,16 +131,17 @@ object Transaction {
     params.mapValues(toNeoType).asJava
 
   private def toNeoType(value: Any): AnyRef = value match {
-    case s: Seq[_]    => s.map(toNeoType).asJava
-    case s: Set[_]    => s.map(toNeoType).asJava
-    case m: Map[_, _] => m.mapValues(toNeoType).asJava
-    case o: Option[_] => o.map(toNeoType).orNull
+    case s: Seq[_]        => s.map(toNeoType).asJava
+    case s: Set[_]        => s.map(toNeoType).asJava
+    case m: Map[_, _]     => m.mapValues(toNeoType).asJava
+    case o: Option[_]     => o.map(toNeoType).orNull
+    case r: Refined[_, _] => toNeoType(r.value)
```

Comment:
I see a big problem here in order to make the refined integration on an module outside `core`. The mappers can be easily moved to another module. But, this part here not.

Any ideas?

Reply:
I fully agree with this being moved to a separate module.
I'm thinking about these two approachs:

**Option 1** 
1. Add a partial function or a sequence of partial functions to `Transaction` as a constructor parameter. So that the `toNeoType` would check if there are anything was injected for overwriting types.
2. Injection part - since the transaction initialization can be done on any level (driver, session, transaction), it seems that that parameter should be passed down to tx from any top level.
I'm currently inclined toward creating an extension method `withRefined` that "adds proper converters to the scope"
```scala
import neotypes.refined.implicits // containes mappers + `withRefined`
...
driver.asScala[Future].withRefined
...
// or
...
driver.session().asScala[Future].withRefined
```
**Option 2 (I don't know if it's possible)** 
1. Add an additional typeclass :`final class Transaction[F[_]](transaction: NTransaction)(implicit F: Async[F], converter: converter: ToNeoTypes[F]) {`
2. (*I'm not completely sure re this, I'm not an expert but I saw something like this but never thought how it's implemented 😁*) Create a DSL that combines typeclasses to have it like this `driver.asScala[Future and RefinedSupport]`

@BalmungSan what do you think? 😁